docs/git_workflow_and_management.md

```diff
@@ -0,0 +1,73 @@
+# Dev Workflow Management
+
+## Gitflow Pattern
+
+Follow a branch + pull-request pattern to create a commit tree structure that matches this structure:
+
+![gitflow structure](../images/git-flow.drawio.svg)
+
+- When starting a new feature/fix/docs/etc. branch, make a small initial commit and push it to Github;
+open a Pull Request immediately so others can track your progress (see what to write in the PR description in Pull Request Conventions).
+- Commit new work to your local branches and regularly push work to the remote; the Pull Request should reflect these changes immediately.
+- To request feedback or help, or when you think your work is ready to merge into the `develop` branch, assign a 'reviewer' to do a **Code Review** (see Code Review Practices).
+- After your work or feature has been *reviewed and approved*, it can be merged into the `develop` branch.
+
+## Branches
+
+### `main` branch
+
+- Any code in the `main` branch should be deployable.
+- Only use Pull Requests to merge into `main` branch.
+- Only ever base merges from the `develop` branch into `main` branch.
+
+### `develop` branch
+
+- Maintain a `develop` For *anything larger than a WIP/experimental* project.
+- All new work should branch out of `develop` – never out of `main` except for `hotfix-*`
+
+### Feature/Tests/Docs branches
+
+- Create new descriptively-named branches off the `develop` branch (only!) for new work, such as `taylor/feature-add-new-payment-types` (see full naming convention in Branch Conventions).
+- Always add new tests to confirm the major behavior of your features (Use HAPPY/BAD/SAD tests).
+- Make sure to test and refactor your feature branches before merging into `develop`
+
+If Github tells you that your PR to merge back into `develop` has conflicts with `develop` (i.e., there have been new, conflicting commits on `develop` since after your branched out), please rebase your work on rebase before trying to merge back into it using a PR.
+
+The rebase restructuring strategy looks like this:
+
+![rebase strategy](../images/git-rebase.drawio.svg)
+
+
+And the corresponding code will be like:
+
+```shell
+# Start on your `feature` (or tests, docs) branch
+git switch develop
+git pull
+
+git switch feature
+git rebase develop
+
+# Rebasing will be interrupted by conflicts
+# (fix all the conflicts and add them to staging)
+# ...
+git add .
```

Comment:
Suggest that if there is a result conflicts in test result.
Or the package version conflict
you could neglect the errors add conflicts file to staged then use like npm I ,or rake spec to override the conflicts

Reply:
Great point, but yes and no to the proposal. For files like `Gemfile.lock` it is better to revert to older version (accept 'current' code in conflict, ignore 'incoming') and rebundle. Same for code coverage (use 'current', ignore 'incoming') and rerun specs. I don't think you should ever add conflicts to staging – it's too risky to leave them in the commit by accident. I'll try to write it this way but welcome to disagree on next commit!

Reply:
there are some issues like vcr may have generated different result, and the conflicts can't be readable, and this time you should use the command to generate a no conflicts one

Reply:
I've left a new `todo` suggestion to have a whole new document just about Conflicts!!